src/big/big_int.cr

```diff
@@ -403,6 +402,284 @@ struct BigInt < Int
+    check_division_by_zero(other)
+
+    if other < 0
+      neg!.unsafe_floored_div!(-other)
+    else
+      unsafe_floored_div!(other)
+    end
+  end
+
+  # Mutating `#tdiv`
+  def tdiv!(other : Int) : BigInt
+    check_division_by_zero(other)
+    unsafe_truncated_div!(other)
+  end
+
+  # Mutating `#unsafe_floored_div`
+  def unsafe_floored_div!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      LibGMP.set_ui(mpz, 1)
+    else
+      LibGMP.fdiv_q(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating `#unsafe_floored_div`
+  def unsafe_floored_div!(other : Int) : BigInt
+    if LibGMP::ULong == UInt32 && (other < Int32::MIN || other > UInt32::MAX)
+      unsafe_floored_div!(other.to_big_i)
+    elsif other < 0
+      LibGMP.fdiv_q_ui(mpz, neg!, other.abs)
+    else
+      LibGMP.fdiv_q_ui(mpz, self, other)
+    end
+
+    self
+  end
+
+  # Mutating `#unsafe_truncated_div`
+  def unsafe_truncated_div!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      LibGMP.set_ui(mpz, 1)
+    else
+      LibGMP.tdiv_q(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating `#unsafe_truncated_div`
+  def unsafe_truncated_div!(other : Int) : BigInt
+    if LibGMP::ULong == UInt32 && (other < Int32::MIN || other > UInt32::MAX)
+      unsafe_truncated_div!(other.to_big_i)
+    elsif other < 0
+      LibGMP.tdiv_q_ui(mpz, neg!, other.abs)
+    else
+      LibGMP.tdiv_q_ui(mpz, self, other)
+    end
+
+    self
+  end
+
+  # Mutating `#%`
+  def mod!(other : Int) : BigInt
+    check_division_by_zero(other)
+
+    if other < 0
+      neg!
+      unsafe_floored_mod!(-other)
+      neg!
+    else
+      unsafe_floored_mod!(other)
+    end
+  end
+
+  # Mutating `#remainder`
+  def remainder!(other : Int) : BigInt
+    check_division_by_zero(other)
+    unsafe_truncated_mod!(other)
+  end
+
+  # Mutating `#unsafe_floored_mod`
+  def unsafe_floored_mod!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      LibGMP.set_ui(mpz, 0)
+    else
+      LibGMP.fdiv_r(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating `#unsafe_floored_mod`
+  def unsafe_floored_mod!(other : Int) : BigInt
+    if other < LibGMP::Long::MIN || other > LibGMP::ULong::MAX
+      unsafe_floored_mod!(other.to_big_i)
+    elsif other < 0
+      LibGMP.fdiv_r_ui(mpz, self, other.abs)
+      neg!
+    else
+      LibGMP.fdiv_r_ui(mpz, self, other)
+    end
+
+    self
+  end
+
+  # Mutating `#unsafe_truncated_mod`
+  def unsafe_truncated_mod!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      LibGMP.set_ui(mpz, 0)
+    else
+      LibGMP.tdiv_r(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating `#unsafe_truncated_mod`
+  def unsafe_truncated_mod!(other : LibGMP::IntPrimitive) : BigInt
+    LibGMP.tdiv_r_ui(mpz, self, other.abs)
+    self
+  end
+
+  # Mutating `#unsafe_truncated_mod`
+  def unsafe_truncated_mod!(other : Int) : BigInt
+    LibGMP.tdiv_r_ui(mpz, self, other.abs.to_big_i)
+    self
+  end
+
+  # Mutating `#~`
+  def com! : BigInt
```

Comment:
I think `#complement!` would be a much clearer name for this method. "com" is just so widely used at the beginning of English words that, even in this specific domain, there's no reason to expect it to mean "complement" without reading the docs or code.